File: JavaScript/11. Strings/Strings-Homework/Strings-Homework/01. ReverseString/ReverseString.js
/*Problem 1. Reverse string

Write a JavaScript function that reverses a string and returns it.*/

function reverseString(string) {
    string = string || '';
    var change = '',
        i = string.length - 1;
    for (; i >= 0; i -= 1) {
        change += string[i];
    }
    return change;
}

reverseString('sample');

File: C# II/06. String and Text Processing/StringAndTextProcessingHomework/02. ReverseString/ReverseString.cs
using System;
using System.Text;

class ReverseString
{
    /*
        2. Reverse string

            Write a program that reads a string, reverses it and prints 
            the result at the console.

        Example:
        input 	output
        sample 	elpmas
    */
    static void Main(string[] args)
    {
        Console.Write("Enter a string: ");
        string input = Console.ReadLine();
        StringBuilder reversedStrb = new StringBuilder();
        for(int i = input.Length-1; i >= 0; i--)
        {
            reversedStrb.Append(input[i]);
        }
        Console.WriteLine("Input: " + input + "  Reversed: " + reversedStrb.ToString());
    }
}

